keps/sig-apps/4563-evacuation-api/README.md

```diff
@@ -0,0 +1,1561 @@
+blocking validating admission webhook, then the evacuation controller will enter into an eviction
+cold loop. To mitigate this we will increment `.status.failedEvictionCounter` and
+`evacuation_controller_evictions` metric.
+
+An evacuator could reconcile the status properly without making any progress. It is thus
+recommended to check `creationTimestamp` of the Evacuations and observe
+`evacuation_controller_active_evacuator` metric to see how much it takes for an evacuator to
+complete the evacuation. This metric can be also used to implement additional alerts.
+
+#### Disruptive Eviction
+
+When using kubectl drain, pods without owning controller and pods with local storage
+(having `emptyDir` volumes) are not evicted by default. We have decided to evict most
+of the pods (except DaemonSet and mirror pods) by default. In the motivation section of the
+[Declarative Node Maintenance KEP](https://github.com/kubernetes/enhancements/pull/4213),
+we can see many administrators override these default settings and many components evict all pods
+indiscriminately. There are also many ways that users use to prevent the eviction;
+PodDisruptionBudgets, validating admission webhooks, or just plain HA. Users who want to protect
+their applications in today's clusters should already be aware that they should be able to handle
+sudden evictions. Therefore, it should be okay for the evacuation controller to evict these pods.
+
+To mitigate the sudden eviction problem, users should use PodDisruptionBudgets or HA.
+
+## Design Details
+
+### Evacuation
+
+We will introduce a new type called `evacuation.coordination.k8s.io`  to enforce the contract
+between the evacuation instigators and the evacuators. This type is a bit similar to
+`leases.coordination.k8s.io` in that it requires multiple actors to synchronize the state. Which in
+our case is the progress of the evacuation.
+
+### Evacuation Instigator
+[Evacuation Instigator](#evacuation-instigator) section provides a general overview.
+
+There can be many evacuation instigators for a single Evacuation.
+
+When an instigator decides that a pod needs to be evacuated, it should create an Evacuation:
+- With a predictable and consistent name in the following format: `${POD_UID}-${POD_NAME_PREFIX}`.
+  The pod name may be only partial as the maximum allowed length of `POD_NAME_PREFIX` is limited
+  to 150 characters.
+- `.spec.podRef` should be set to fully identify the pod. Similar to the name, the UID should be
+  specified to ensure that we do not evacuate a pod with the same name that appears immediately
+  after the previous pod is removed.
+- `.spec.progressDeadlineSeconds` should be set to a reasonable value. It is recommended to leave
+  it at the default value of 1800 (30m) to allow for potential evacuator disruptions.
+- `.spec.evacuators` value will be resolved on the Evacuation admission from the pod and should
```

Comment:
I have gotten this far, but I really don't know what this is describing.   This is a very long and detailed KEP, but I think it would really benefit from an overview with less details.  E.g.

"""
Pods carry new information about which "evacuators" (described later) are involved in its lifecycle.   Multiple evacutation initiators can coordinate, via a new "Evacuation" API object, to indicate the need for a pod to be removed from a node.  This new object  allows the evacuators to take action before the pod is removed, and...
"""

Reply:
I have improved the summary of the proposal and added links to practical examples: https://github.com/atiratree/kube-enhancements/blob/evacuation-api/keps/sig-apps/4563-eviction-request-api/README.md#proposal